app/controllers/concerns/foreman/controller/smart_proxy_auth.rb

```diff
@@ -7,16 +7,25 @@ module Foreman::Controller::SmartProxyAuth
   module ClassMethods
     def add_smart_proxy_filters(actions, options = {})
       skip_before_filter :require_login, :only => actions
-      skip_before_filter :require_ssl, :only => actions
       skip_before_filter :authorize, :only => actions
       skip_before_filter :verify_authenticity_token, :only => actions
       skip_before_filter :set_taxonomy, :only => actions
       skip_before_filter :session_expiry, :update_activity_time, :only => actions
       before_filter(:only => actions) { require_smart_proxy_or_login(options[:features]) }
       attr_reader :detected_proxy
+      define_method("require_ssl_with_dont?") do
+        send("require_ssl?") unless actions
```

Comment:
`unless actions` will always be false.. shouldn't that check if `actions.include?(the current action)`?